x/ibc/README.md

```diff
@@ -0,0 +1,229 @@
+# IBC Core Implementation Specification
+
+## State Structure
+
+### ConnID
+
+// TODO: %s ConnID ChainID
+`ConnID` is the local name for a connected chain. When a conn is established between two chains, each of them subjectively name the other. Is is done by storing a mapping from the `ConnID` to the `ChainConfig`. 
+
+* Proposal1: IP-style format
+
+A `ConnID` is length 4 of byte array. 
+
+NOTE: due to the state structure, the length of `ConnID`s should not be same with the `PortID`s'. 
+
+There can be a nameservice for `ConnID`s, providing human-readable aliases.
+
+* Proposal2: domain-style format
+
+A `ConnID` is variadic length of alphanumeric slice, where its length is prefixed.
+
+There can be a nameservice for `ConnID`s, where the users can bid for the names.
+
+`ConnID`s does not have to be uniform on the network. Naming is subjective on the chains.
```

Comment:
This is one of the most important security considerations. 

The proposition is that if you are holding an asset on a chain that is honestly following the IBC protocol, a dishonest chain cannot maliciously choose a ConnID to the conceals the counterparty relationship with the dishonest chain from the asset holder.

This should be explicitly stated.